symengine/polynomial.cpp

```diff
@@ -374,30 +374,31 @@ UnivariatePolynomial::UnivariatePolynomial(const RCP<const Symbol> &var,
     unsigned int deg = 0;
```

Comment:
Here, instead of relying on the overloading, I think it should be explicitly `dict_.dict_ = {}`. I think the naming is causing a bit of a confusion.